### PR TITLE
feat(checkbox-input): support isReadOnly prop

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Addresses one of the cases [here](https://github.com/commercetools/ui-kit/issues/695)

### CODE REVIEW

File: src/components/inputs/checkbox-input/checkbox.js [hunk 1]
```diff
@@ -11,48 +11,50 @@ const Input = styled.input`
...
-      />
-    );
-  }
-}
+  }, [props.isIndeterminate]);
+
+  const { onChange } = props;
+
+  const handleChange = React.useCallback(
+    event => !props.isReadOnly && onChange && onChange(event),
```
Comment (montezume) [inline src/components/inputs/checkbox-input/checkbox.js]: Why use JS? Why isn't setting `readOnly` on the input enough?

https://stackoverflow.com/questions/155291/can-html-checkboxes-be-set-to-readonly

File: src/components/inputs/checkbox-input/checkbox-input.styles.js [hunk 2]
```diff
@@ -33,6 +33,21 @@ const getCheckboxWrapperStyles = (props, theme) => {
       `,
     ];
   }
+  if (props.isReadOnly) {
+    return [
+      baseStyles,
+      css`
+        svg [id$='borderAndContent'] > [id$='content'] {
+          fill: ${overwrittenVars[designTokens.fontColorForInputWhenReadonly]};
```
Comment (montezume) [inline src/components/inputs/checkbox-input/checkbox-input.styles.js]: same look as for RadioInputs when readOnly

File: src/components/inputs/checkbox-input/checkbox.js (hunk 1 above)
Comment (tdeekens) [reply src/components/inputs/checkbox-input/checkbox.js]: Thanks for sharing. I often wanted to ask UX "what's the different line in communication towards the user between readonly and disabled? When use which?"

**Comment by montezume** [comment]
@tdeekens yeah, I like to think about it like 

readonly: submitted in form
disabled: not submitted in form

[ref](https://stackoverflow.com/questions/1355728/values-of-disabled-inputs-will-not-be-submitted#comment23440330_1355728)

**Pull request merged**